Bundle/BusinessEntityPageBundle/Helper/BusinessEntityPageHelper.php

```diff
@@ -242,28 +246,31 @@ public function getIdentifierPositionInUrl(BusinessEntityPagePattern $bepPattern
      * @param  \ReflectionClass $refClass
      * @return View
      */
-    public function guessBestViewForEntity($refClass)
+    public function guessBestPatternIdForEntity($refClass, $entityId)
     {
         $pattern = null;
         $classname = $refClass->name;
         $businessEntity = $this->businessEntityHelper->findByEntityClassname($classname);
         if ($businessEntity) {
-            $patterns = $this->entityManager->getRepository('VictoireBusinessEntityPageBundle:BusinessEntityPagePattern')->findByBusinessEntityName($businessEntity->getName());
-            if (count($patterns) > 0) {
-                $pattern = array_pop($patterns);
-            }
+            $parameters = array(
+                'entityId' => $entityId,
+                'entityNamespace' => $refClass->getName()
+            );
+
+            $viewReference = $this->viewCacheHelper->getReferenceByParameters($parameters);
+
         }
 
-        if (!$pattern) {
+        if (!$viewReference) {
             $parentRefClass = $refClass->getParentClass();
             if ($parentRefClass) {
-                $pattern = $this->guessBestViewForEntity($parentRefClass);
+                $pattern = $this->guessBestPatternIdForEntity($parentRefClass, $entityId);
```

Comment:
This variable is never used because we return `$viewReference['patternId']`, I guess there is something to do.